### PR TITLE
feat: Adds support on direction RTL in UI 

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes https://github.com/chatwoot/chatwoot/issues/686

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
https://www.loom.com/share/53e305e7d58e4576ae6014c7dcc0554d

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/components/CustomAttribute.vue [hunk 1]
```diff
@@ -6,6 +6,7 @@
           <input
             v-model="editedValue"
             class="checkbox"
+            :class="{ 'checkbox--rtl': isRTLView }"
```
Comment (scmmishra) [inline app/javascript/dashboard/components/CustomAttribute.vue:9]: Do we need this? We can write the rule to apply any style if there's `app-rtl--wrapper` on its parents?

.app-rtl--wrapper .checkbox {
   margin-left: var(--space-small);
}

File: app/javascript/dashboard/components/buttons/ResolveAction.vue [hunk 2]
```diff
@@ -261,6 +261,17 @@ export default {
   display: flex;
   align-items: center;
   justify-content: flex-end;
+
+  .button-group {
+    clip-path: inset(0% 0% 0% 0% round var(--border-radius-normal));
```
Comment (scmmishra) [inline app/javascript/dashboard/components/buttons/ResolveAction.vue:266]: Why the clip-path?

Comment (iamsivin) [reply app/javascript/dashboard/components/buttons/ResolveAction.vue]: Updated.

File: app/javascript/dashboard/assets/scss/_utility-helpers.scss [hunk 3]
```diff
@@ -10,8 +10,18 @@
   margin-right: var(--space-smaller);
 }
 
+.margin-left-smaller {
```
Comment (nithindavid) [inline app/javascript/dashboard/assets/scss/_utility-helpers.scss:13]: We have `margin-1` equivalent available in https://get.foundation/sites/docs/prototyping-utilities.html; please use it.

Comment (iamsivin) [reply app/javascript/dashboard/assets/scss/_utility-helpers.scss]: Done https://github.com/chatwoot/chatwoot/pull/6369/commits/158dc29409d3099dfb8e4268cfad6e9447e76fe3

**Comment by hadifarnoud** [comment]
It's mostly good but I found a few issues:
on 0:22
Screenshot 1401-11-25 at 14 59 54

the padding-left should be changed to padding-right for green labels next to text. 
on 0:36
Screenshot 1401-11-25 at 15 00 26

text bubbles look like they are aligned left. they should have `direction:rtl` and `text-align:right`. you can remove text-align:right and it just works as the default for rtl direction is set to right anyway. unless you are using some sort of auto direction method, this should be fixed.

0:45
Screenshot 1401-11-25 at 15 16 26

padding issue between user thumb and text. it should be changed to padding left.


I did not see a test for text editors and any other field you can type text in. for testing purpose you can type/copy phrase below. it must show up exactly in the same order you can see here

تست فارسی و English با هم

**Comment by iamsivin** [comment]
> It's mostly good but I found a few issues:
> 
> تست فارسی و English با هم

@hadifarnoud Thanks for taking a look at the PR. This issue was fixed in this PR https://github.com/chatwoot/chatwoot/pull/6430. The loom video is outdated. Will update it with latest changes

File: app/javascript/dashboard/components/widgets/conversation/MessagesView.vue [hunk 4]
```diff
@@ -416,7 +416,7 @@ export default {
 
     .reply-box .reply-box__top {
       position: relative;
-      min-height: 44rem;
+      height: 50vh;
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/MessagesView.vue:419]: Are we sure about this change? @iamsivin just verify this does not break on multiple screen sizes

File: app/javascript/dashboard/routes/dashboard/settings/canned/Index.vue [hunk 5]
```diff
@@ -2,7 +2,7 @@
   <div class="column content-box">
     <woot-button
       color-scheme="success"
-      class-names="button--fixed-right-top"
+      class-names="button--fixed-top"
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/canned/Index.vue:5]: I see a lot of changes with `button--fixed-right-top` changed to `button--fixed-top`

Just verify if that it's working correctly for multiple screens, you can use https://responsively.app/ or https://polypane.app/ to test these out quickly

**Review by scmmishra — APPROVED** [review]
Approving these changes, I would suggest testing it on staging once. From a code point of view, this seems good to go

File: app/javascript/dashboard/components/widgets/conversation/MessagesView.vue (hunk 4 above)
Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/conversation/MessagesView.vue:419]: @scmmishra Here is the quick look related to this changes https://www.loom.com/share/9eaf3818f47c4e6dba267c02b861e6c7

File: app/javascript/dashboard/routes/dashboard/settings/canned/Index.vue (hunk 5 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/settings/canned/Index.vue:5]: This class is used for this button in the settings screens.
image

Here is the loom video.
https://www.loom.com/share/8715173e0d7f4d3a87367f7c4c2292c4

File: app/javascript/dashboard/components/CustomAttribute.vue [hunk 6]
```diff
@@ -271,6 +271,19 @@ export default {
...
+.input-group {
+  margin-top: var(--space-smaller);
+  margin-bottom: var(--space-smaller);
+}
+
+.input-group-button {
+  margin: 0 var(--space-smaller);
+
+  button {
+    border-radius: var(--border-radius-normal) !important;
```
Comment (nithindavid) [inline app/javascript/dashboard/components/CustomAttribute.vue:283]: Rewrite rule

**Comment by iamsivin** [comment]
Closing this PR as splitted this PR into multiples and merged to develop.
https://github.com/chatwoot/chatwoot/pull/6515, https://github.com/chatwoot/chatwoot/pull/6516, https://github.com/chatwoot/chatwoot/pull/6517, https://github.com/chatwoot/chatwoot/pull/6518, https://github.com/chatwoot/chatwoot/pull/6519, https://github.com/chatwoot/chatwoot/pull/6521, https://github.com/chatwoot/chatwoot/pull/6524, https://github.com/chatwoot/chatwoot/pull/6525, https://github.com/chatwoot/chatwoot/pull/6526, https://github.com/chatwoot/chatwoot/pull/6531, https://github.com/chatwoot/chatwoot/pull/6537

**Comment by hadifarnoud** [comment]
cool. how can I try it locally?

**Comment by iamsivin** [comment]
> cool. how can I try it locally?

@hadifarnoud You can try it locally in the latest develop branch. 
**Steps**
1. By going to account settings 
2. Change the language that supports RTL (supported language code - 'ar', 'as', 'fa', 'he', 'ku', 'ur')
3. And update the settings. Then the direction will be changed.

**Comment by hadifarnoud** [comment]
what branch I should check out?

**Comment by iamsivin** [comment]
> what branch I should check out?

You can test it in the default `develop` branch.

**Pull request closed without merging**